repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -166,10 +167,16 @@ def prepare_target_userspace(context, userspace_dir, enabled_repos, packages):
         try:
             context.call(cmd, callback_raw=utils.logging_handler)
         except CalledProcessError as exc:
-            raise StopActorExecutionError(
-                message='Unable to install RHEL {} userspace packages.'.format(target_major_version),
-                details={'details': str(exc), 'stderr': exc.stderr}
-            )
+            message = 'Unable to install RHEL {} userspace packages.'.format(target_major_version)
+            details = {'details': str(exc), 'stderr': exc.stderr}
+
+            # If a proxy was set in dnf config, it should be the reason why dnf
+            # failed since leapp does not support updates behind proxy yet.
+            for manager_info in api.consume(PkgManagerInfo):
+                if manager_info.dnf_proxy:
+                    details['details'] = "Cannot download packages when behind proxy."
```

Comment:
The message provided should be more user friendly, e.g. _DNF failed to install userspace packages, likely due to the proxy configuration detected in the YUM/DNF configuration._ This should be discussed with @pirat89 or @Rezney.